physics/cires_ugwp_module_v1.F90

```diff
@@ -0,0 +1,666 @@
+    deallocate( rfdis, rfdist)
+    deallocate(ugwp_taulat,  ugwp_qbolat)
+    deallocate(tau_limb, uzmf_merra)
+    deallocate(days_limb,  days_merra,  pmb127)
+ 
+   end subroutine cires_ugwp_finalize
+
+!
+!
+!
+!
+    subroutine calendar_ugwp(yr, mm, dd, ddd_ugwp)
+!
+! computes day of year to get tau_limb forcing written with 1-day precision
+!    
+    implicit none
+    integer, intent(in) :: yr, mm, dd
+    integer :: ddd_ugwp
+    
+    integer ::  iw3jdn
+    integer :: jd1, jddd
+    jd1     = iw3jdn(yr,1,1)
+    jddd    = iw3jdn(yr,mm,dd)
+    ddd_ugwp = jddd-jd1+1
+    
+    end subroutine calendar_ugwp
+    
+    
+    subroutine cires_indx_ugwp (npts, me, master, dlat,j1_tau,j2_tau, w1_j1tau, w2_j2tau, &
+                                j1_qbo,j2_qbo, w1_j1qbo, w2_j2qbo, dexp_latqbo  )   
+	 
+    implicit none
+!    
+! ntau_d1y, ntau_d2t, nqbo_d1y, nqbo_d2z, nqbo_d3t
+! ugwp_taulat(:),  ugwp_qbolat(:), ugwp_merlat(:)
+!
+      integer    ::   npts, me, master
+      integer, dimension(npts) ::  j1_tau,j2_tau, j1_qbo, j2_qbo
+      real ,   dimension(npts) ::  dlat, w1_j1tau, w2_j2tau, w1_j1qbo, w2_j2qbo
+      real ,   dimension(npts) ::  dexp_latqbo  
+      real    :: widqbo2, xabs          
+!
+      integer i,j, j1, j2
+!      
+! weights for tau_limb  w1_j1tau, w2_j2tau
+!
+      do j=1,npts
+        j2_qbo(j) = nqbo_d1y
+        do i=1, nqbo_d1y
+          if (dlat(j) < ugwp_qbolat(i)) then
+            j2_qbo(j) = i
+            exit
+          endif
+        enddo
+	
+        
+        j2_qbo(j) = min(j2_qbo(j),nqbo_d1y)
+        j1_qbo(j) = max(j2_qbo(j)-1,1)	
+	
+        if (j1_qbo(j) /= j2_qbo(j) ) then
+          w2_j2qbo(j) = (dlat(j)  - ugwp_qbolat(j1_qbo(j))) &
+                 / (ugwp_qbolat(j2_qbo(j))-ugwp_qbolat(j1_qbo(j)))
+       	 
+        else
+          w2_j2qbo(j) = 1.0
+        endif
+          w1_j1qbo(j) = 1.0 - w2_j2qbo(j)	
+	  
+!
+      enddo	 
+!	   
+! weights for tau_limb  w1_j1tau, w2_j2tau
+!
+      do j=1,npts
+        j2_tau(j) = ntau_d1y
+        do i=1,ntau_d1y
+          if (dlat(j) < ugwp_taulat(i)) then
+            j2_tau(j) = i
+            exit
+          endif
+        enddo
+	
+      
+        j2_tau(j) = min(j2_tau(j),ntau_d1y)
+        j1_tau(j) = max(j2_tau(j)-1,1)	
+	
+        if (j1_tau(j) /= j2_tau(j) ) then
+          w2_j2tau(j) = (dlat(j)  - ugwp_taulat(j1_tau(j))) &
+                 / (ugwp_taulat(j2_tau(j))-ugwp_taulat(j1_tau(j)))
+       	 
+        else
+          w2_j2tau(j) = 1.0
+        endif
+          w1_j1tau(j) = 1.0 -	w2_j2tau(j)	
+
+      enddo
+ 		widqbo2 =1./widqbo/widqbo     
+       do j=1,npts         
+       	    dexp_latqbo(j) =0.		 		 
+	    xabs =abs(dlat(j))
+	if (xabs .le. latqbo) then
+	    dexp_latqbo(j) = exp(-xabs*xabs*widqbo2)
+	    if (xabs .le. 4.0 ) dexp_latqbo(j)  =1.
+!	    print *, ' indx_ugwp dexp=', dexp_latqbo(j), nint(dlat(j))
+        endif		 
+       enddo     
+      
+      if (me == master ) then
```

Comment:
Is this called every time step? Or is this a debugging routine that is not active by default?

Reply:
As far as I can tell, this subroutine isn't called.  The cires_ugwp_v1 code is still under development.  If possible, can we defer this comment until later?